frontend/model/contracts/voting/rules.js

```diff
@@ -73,3 +83,39 @@ const rules = {
 export default rules
 
 export const ruleType = unionOf(...Object.keys(rules).map(k => literalOf(k)))
+
+/**
+ *
+ * @example ('disagreement', 2, 1) => 2
+ * @example ('disagreement', 5, 1) => 3
+ * @example ('disagreement', 7, 10) => 7
+ * @example ('disagreement', 20, 10) => 10
+ *
+ * @example ('percentage', 0.5, 3) => 0.5
+ * @example ('percentage', 0.1, 3) => 0.33
+ * @example ('percentage', 0.1, 10) => 0.2
+ * @example ('percentage', 0.3, 10) => 0.3
+ */
+export const getThresholdAdjusted = (rule, threshold, groupSize) => {
+  const groupSizeVoting = Math.max(3, groupSize) // 3 = minimum groupSize to vote
+
+  if (rule === RULE_DISAGREEMENT) {
+    // Limit number of maximum "no" votes to group size
+    return Math.min(groupSizeVoting - 1, threshold)
+  }
+
+  if (rule === RULE_PERCENTAGE) {
+    // Minimum threshold correspondent to 2 "yes" votes
+    const minThreshold = 2 / groupSizeVoting
+    return Math.max(minThreshold, threshold)
+  }
```

Comment:
Always avoid `if`s if you can.

Here's another great place to use a dictionary/map/object instead:

```js
return {
  [RULE_DISAGREEMENT]: function (...) { .. }
  // ...
}[rule]()
```

This has several advantages:

1. It is more concise and easier to read
2. It is much faster
3. It will throw an exception automatically if the wrong rule type is passed in

Reply:
Updated. Here and it other places with similar if usage.